File: GiftAppWebAPI/Models/Application/Gift.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace GiftAppWebAPI.Models.Application
{
    public class Gift
    {
        public int Id { get; set; }
        public string GiverUsername { get; set; }
        public string ReceiverUsername { get; set; }
        public string Name { get; set; }
        public int Quantity { get; set; }
        public bool IsNew { get; set; }
    }
}

File: GiftApp/src/app/_models/gift.ts
export class Gift {
    public id: number;
    public giverUsername: string;
    public isNew: boolean;
    public name: string;
    public quantity: number;
    public receiverUsername: string;

    constructor(id: number, giverUsername: string, isNew: boolean, name: string, quantity: number, receiverUsername: string) {
        this.id = id;
        this.giverUsername = giverUsername;
        this.isNew = isNew;
        this.name = name;
        this.quantity = quantity;
        this.receiverUsername = receiverUsername;
    }
}
